File: c/questao12.c
#include <stdio.h>


int primo(int number) {

    if (number < 2) return 0;

    int i;
    for (i = 2; i <= number / 2; i++) {
        if (number % i == 0) {
            return 0;
        }
    }

    return 1;
}
int main() {

    int number;

    printf("Digite o número que deseja testar: ");
    scanf("%d", &number);

    if (primo(number)){
        printf("É PRIMO!\n");
    } else{
        printf("NÃO É PRIMO!\n");
    }
    return 0;
}

File: python/questao12.py
def primo(number):
    if number < 2:
        return False

    for i in range(2, int(number / 2) + 1):
        if number % i == 0:
            return False

    return True

def main():
    number = int(input("Digite o número que deseja testar: "))

    if primo(number):
        print("É PRIMO!")
    else:
        print("NÃO É PRIMO!")

if __name__ == "__main__":
    main()
